src/db.rs

```diff
@@ -38,315 +40,410 @@ impl Default for Connection {
+
+    let con_row = sqlx::query(&new_query_str)
+        .bind(uuidstr)
+        .fetch_one(&dbpool)
+        .await?;
+    Ok(Connection {
+        policy: con_row.try_get::<i32, _>(0)? as u32,
+        fw_api_major: con_row.try_get::<i32, _>(1)? as u32,
+        fw_api_minor: con_row.try_get::<i32, _>(2)? as u32,
+        fw_build_id: con_row.try_get::<i32, _>(3)? as u32,
+        launch_description: con_row.try_get::<String, _>(4)?,
+        fw_digest: con_row.try_get::<String, _>(5)?,
     })
 }
 
-pub fn delete_policy(pid: &u64) -> Result<()> {
-    let mut dbconn = get_dbconn()?;
+pub async fn delete_connection(uuid: Uuid) -> Result<Uuid> {
+    let uuidstr = uuid.as_hyphenated().to_string();
 
-    let mqstr = "DELETE from policy WHERE id = ?";
+    let dbpool = get_dbpool().await?;
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (&pid,))?;
-    trnsx.commit()?;
-    Ok(())
+    let query_str = "DELETE from conn_bundle WHERE id = ?";
+    let new_query_str = replace_binds(dbpool.any_kind(), query_str);
+
+    sqlx::query(&new_query_str)
+        .bind(uuidstr)
+        .execute(&dbpool)
+        .await?;
+    Ok(uuid)
 }
 
-pub fn delete_connection(uuid: Uuid) -> Result<Uuid> {
-    let mut dbconn = get_dbconn()?;
+pub async fn insert_policy(policy: &policy::Policy) -> Result<u64> {
+    let allowed_digests_json = serde_json::to_string(&policy.allowed_digests)?;
+    let allowed_policies_json = serde_json::to_string(&policy.allowed_policies)?;
+    let allowed_build_ids_json = serde_json::to_string(&policy.allowed_build_ids)?;
 
-    let mqstr = "DELETE from conn_bundle WHERE id = ?";
+    let dbpool = get_dbpool().await?;
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (uuid.as_hyphenated().to_string(),))?;
-    trnsx.commit()?;
-    Ok(uuid)
+    let db_type = env::var("KBS_DB_TYPE").expect("KBS_DB_TYPE not set");
+    let mut query_str = if db_type == "sqlite" {
+        String::from("INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids, create_date, valid) VALUES(?, ?, ?, ?, ?, DATE('now'), 1)")
+    } else {
+        String::from("INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids, create_date, valid) VALUES(?, ?, ?, ?, ?, NOW(), 1)")
+    };
+    //let mut query_str = String::from("INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids, create_date, valid) VALUES(?, ?, ?, ?, ?, NOW(), 1)");
+
+    if dbpool.any_kind() == AnyKind::MySql || dbpool.any_kind() == AnyKind::Sqlite {
+        let last_insert_row = sqlx::query(&query_str)
+            .bind(allowed_digests_json)
+            .bind(allowed_policies_json)
+            .bind(policy.min_fw_api_major as i64)
+            .bind(policy.min_fw_api_minor as i64)
+            .bind(allowed_build_ids_json)
```

Comment:
Can we store a query that is party built as a variable. As in `blah = sqlx::query(&query_str).bind(foo).bind(bar)` so that we could conditionally add stuff to blah?

Reply:
I don't think that this is possible as written.  It is my understanding that we need a query string that has at least one placeholder for bind to work.  If we wanted to do this, we might make the code more complicated.  I can take another look at this, but would it be possible to do it in the next PR?